### PR TITLE
port apply fsm step 3

### PR DESCRIPTION
**Pull request by coocood**
Also replaced pb type `RaftLocalState` and `RaftApplyState` to simply struct type to reduce clone cost.

### CODE REVIEW

File: tikv/raftstore/fsm_apply.go [hunk 1]
```diff
@@ -446,23 +615,121 @@ func (d *applyDelegate) updateMetrics(aCtx *applyContext) {
...
+	wb.Set(ApplyStateKey(d.region.Id), d.applyState.Marshal())
 }
 
 func (d *applyDelegate) handleRaftEntryNormal(aCtx *applyContext, entry *eraftpb.Entry) applyResult {
-	return applyResult{} // TODO: stub
+	index := entry.Index
+	term := entry.Term
+	if len(entry.Data) > 0 {
+		cmd := new(raft_cmdpb.RaftCmdRequest)
+		err := cmd.Unmarshal(entry.Data)
```
Comment (zhangjinpeng1987) [inline tikv/raftstore/fsm_apply.go]: Should check the term and index after unmarshal from bytes.

File: tikv/raftstore/fsm_apply.go [hunk 2]
```diff
@@ -446,23 +615,121 @@ func (d *applyDelegate) updateMetrics(aCtx *applyContext) {
...
+		panic("unreachable")
+	}
 }
 
 func (d *applyDelegate) findCallback(index, term uint64, isConfChange bool) Callback {
-	return nil // TODO: stub
+	regionID := d.region.Id
+	peerID := d.id
+	if isConfChange {
+		for {
```
Comment (zhangjinpeng1987) [inline tikv/raftstore/fsm_apply.go]: Seems this loop is no need.

File: tikv/raftstore/fsm_apply.go (hunk 1 above)
Comment (coocood) [reply tikv/raftstore/fsm_apply.go]: but tikv doesn't check term and index.

Comment (zhangjinpeng1987) [reply tikv/raftstore/fsm_apply.go]: in parse_data_at.

**Pull request merged**